src/core/providers/accounts.ts

```diff
@@ -115,3 +97,26 @@ export function createHDWalletProvider(
     return accountProvider.send(method, params);
   });
 }
+
+export function createFromProvider(provider: IEthereumProvider, from?: string) {
+  let account: string | undefined = from;
+
+  return wrapSend(provider, async (method: string, params: any[]) => {
+    if (account === undefined) {
+      const accounts = await provider.send("eth_accounts");
```

Comment:
I wouldn't put this here. As you may want to connect to a node without any account to interact in other ways. For example, you may wan't to just download historic data. Also, it could be a method.

Reply:
I'll also figure out if `eth_call` fails if `from` is missing and there aren't accounts in the node.

Reply:
It doesn't fail. So the logic should be different for `eth_call` and `eth_sendTransaction`. Only the latter should throw.